address/address.go

```diff
@@ -129,3 +132,78 @@ func ToBech32(bc *Bech32) (string, error) {
 
 	return bech, nil
 }
+
+// FromBlech32 decodes a blech32 encoded string, returning the human-readable
+// part and the data part excluding the checksum.
+func (bc *Bech32) FromBlech32(address string) (*Blech32, error) {
```

Comment:
shouldn't the type be `*Blech32`?
```suggestion
func (bc *Blech32) FromBlech32(address string) (*Blech32, error) {
```